### PR TITLE
Migrate to GitHub Actions for CI

### LINKED ISSUE #771: Migrate to GitHub Actions for CI
I'd like to migrate from Travis to GitHub actions.

Anyone wanna take a stab at this before I mark it as "In Progress", please feel free

### PR DESCRIPTION
**Pull request by nayaabkhan**
Thank you for contributing to Reach UI! Please fill in this template before submitting your PR to help us process your request more quickly.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] ~Test the change in your own code (Compile and run).~
- [ ] ~Add or edit tests to reflect the change (Run with `yarn test`).~
- [ ] ~Add or edit Storybook examples to reflect the change (Run with `yarn start`).~
- [x] Ensure formatting is consistent with the project's Prettier configuration.
- [x] Add documentation to support any new features.

This pull request:

- [ ] Creates a new package
- [ ] Fixes a bug in an existing package
- [ ] Adds additional features/functionality to an existing package
- [ ] Updates documentation or example code
- [x] Other

Fixes #771.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+      - name: Run Tests
+        run: yarn test
+
+  release:
+    needs: test
+    name: Release
+    runs-on: ubuntu-latest
+    env:
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (nayaabkhan) [inline .github/workflows/ci.yml:43]: @chaance Needs attention: A [GitHub secret](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-organization) should be created with the value of `NPM_TOKEN` you use.

File: scripts/deploy-website.sh [hunk 2]
```diff
@@ -21,10 +21,10 @@ cd $tmp_dir
 git add -A
 git commit \
   --allow-empty \
-  --author "Travis CI <travis-ci@reacttraining.com>" \
+  --author "GitHub CI <github-ci@reacttraining.com>" \
```
Comment (nayaabkhan) [inline scripts/deploy-website.sh:24]: @chaance **Needs attention:** I've made changes here for consistency, but I'd recommend creating a CI agnostic address like `ci@reacttraining.com` instead.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Checkout
+        run: yarn --frozen-lockfile
+
+      - name: Build Packages
+        run: yarn build
+
+      - run: echo "Deploying website to https://reach.tech/"
+      - run: openssl aes-256-cbc -K $encrypted_e12d3fc86b46_key -iv $encrypted_e12d3fc86b46_iv -in website-deploy-key.enc -out website-deploy-key -d
```
Comment (nayaabkhan) [inline .github/workflows/ci.yml:98]: **Needs attention:** I'm not familiar with this, possibly they're also set on Travis and need to be migrated to GitHub Secrets. If so, should be updated to `secrets.*` format as well.

**Comment by chaance** [comment]
This is great, thanks so much! I'll try to take a look at this and get the outstanding needs resolved this weekend.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (chaance) [reply .github/workflows/ci.yml:98]: Trying to pin down these values, but once I have these I'll update this and get it merged. In the mean time we can replace `$encrypted_e12d3fc86b46_key` with `${{ secrets.encrypted_e12d3fc86b46_key }}` (same for the iv key).

Comment (nayaabkhan) [reply .github/workflows/ci.yml]: Done ✅

**Pull request merged**